c7n/resolver.py

```diff
@@ -40,12 +41,22 @@ def resolve(self, uri):
         else:
             # TODO: in the case of file: content and untrusted
             # third parties, uri would need sanitization
-            fh = urlopen(uri)
-            contents = fh.read().decode('utf-8')
-            fh.close()
+            req = Request(uri, headers={"Accept-Encoding": "gzip"})
+            response = urlopen(req)
```

Comment:
Can this be done with a context manager?
```
with urlopen(req) as response:
    contents = self.handle_content_encoding(response)
```
Or does that create compatibility issues?

Reply:
Updated.

Reply:
doesn't look updated